### ISSUE TITLE
Removing non-existent intermediate sub-key should not throw an exception

### ISSUE THREAD

**Issue by bfink13** [bwaldvogel/mongo-java-server#64]
The current behavior for removing a sub-key is this:
```
Object subObject = getFieldValueListSafe(document, mainKey);
if (subObject instanceof Document || subObject instanceof List<?>) {
   return removeSubdocumentValue(subObject, subKey, matchPos);
} else {
   throw new MongoServerException("failed to remove subdocument");
}
```
With the code above, if an intermediate sub-key is missing, an exception will be thrown. It is perfectly reasonable that an intermediate sub-key does not exist on the document in question. If it doesn't, the operation (remove or rename) should be ignored for that document.

**Comment by bwaldvogel** [comment]
Could you provide a test case?

**Comment by bfink13** [comment]
Yep, working on it as we speak.

**Comment by bwaldvogel** [comment]
Maybe even willing to submit a PR to fix it?

**Comment by bfink13** [comment]
Yep :) Found one more weird case that I'm trying to iron out...

**Comment by bwaldvogel** [comment]
I’ve just released version `1.15.0` which contains this fix.

**Comment by bfink13** [comment]
Thank you!
